api_tests/files/views/test_file_detail.py

```diff
@@ -705,6 +712,73 @@ def test_load_and_property(self, app, user, file):
             expect_errors=True, auth=user.auth,
         ).status_code == 405
 
+    def test_retracted_registration_file(self, app, user, file_url, file):
+        resource = RegistrationFactory(is_public=True)
+        retraction = resource.retract_registration(
+            user=resource.creator,
+            justification='Justification for retraction',
+            save=True,
+            moderator_initiated=False
+        )
+
+        approval_token = retraction.approval_state[resource.creator._id]['approval_token']
+        retraction.approve(user=resource.creator, token=approval_token)
+        resource.save()
```

Comment:
fun fact, we have an internal shorthand for this:
`retraction.accept()` with no Auth will skip the required approval steps.